extensions/azurecore/src/azureResource/azure-resource.d.ts

```diff
@@ -21,8 +21,33 @@ export namespace azureResource {
 		readonly treeItem: TreeItem;
 	}
 
-	export interface AzureResourceSubscription {
+	export interface AzureResource {
+		name: string;
+	}
+
+	export interface AzureSqlResource extends AzureResource {
+		loginName: string;
+	}
+
+	export interface AzureResourceSubscription extends AzureResource {
 		id: string;
 		name: string;
 	}
+
+	export interface AzureResourceResourceGroup extends AzureResource {
+		subscriptionId: string;
+	}
+
+	export interface AzureResourceDatabase extends AzureSqlResource {
+		serverName: string;
+		serverFullName: string;
+	}
+
+	export interface AzureResourceDatabaseServer extends AzureSqlResource {
+		id?: string;
```

Comment:
Should this be optional? Usually id has to be there.

Reply:
I went back and forth on this. There's no absolute guarantee from the API that it'll have an id so I usually prefer to play it safe and assume it may be undefined.

But if it is undefined that indicates a bigger issue so maybe I'll just have it be required for now and just make sure we have something in place to notify the user something went wrong if this doesn't exist.

Reply:
Yeah I think relying on IDs is pretty common practice. We should probably just not accept anything without an ID (?)

Reply:
Yeah fair enough - changed to be required.